module/zfs/dsl_deadlist.c

```diff
@@ -843,15 +876,28 @@ dsl_deadlist_merge(dsl_deadlist_t *dl, uint64_t obj, dmu_tx_t *tx)
 	}
 
 	mutex_enter(&dl->dl_lock);
+	for (zap_cursor_init(&pzc, dl->dl_os, obj), i = 0;
+	    (perror = zap_cursor_retrieve(&pzc, &pza)) == 0 && i < 128;
```

Comment:
Why this is limited to prefetching 128 deadlists ahead is worth a comment (here and in `dsl_deadlist_move_bpobj()`).  Even if it's just that this was what was experimentally determined to be sufficient.

Reply:
It is a number coming from tests on a 64-wide HDD pool, where the return of further prefetch is mostly diminishing.  Added comments.